### PR TITLE
various integration test improvements

### LINKED ISSUE #731: Write syscall test should fail on very large write
We should add a test for a very large write (larger than the size of the buffer) that fails and returns `EMSGSIZE`.

### LINKED ISSUE #747: Dynamic output from integration tests
Currently, our [integration tests allow us to specify](https://github.com/enarx/enarx/blob/master/integration-tests/tests/common/mod.rs#L71-L78) the input to `stdin` and the expected output on `stdout` and `stderr`. However, this presumes that the outputs are deterministic. https://github.com/enarx/enarx/pull/681 breaks that assumption.

When testing `clock_gettime` we should have the payload print the current time to `stdout` and then have the test harness validate that the output is within an acceptable delta from its own version of `clock_gettime`. Therefore, `run()` function for the tests should be made to return the outputs as `Vec<u8>` and the `clock_gettime` test should be rewritten to handle this.

- [x] Make `run()` return the outputs.
- [x] Fix up existing tests.
- [x] Rewrite `clock_gettime` to output the time, evaluated by the harness.

### PR DESCRIPTION
**Pull request by npmccallum**
Resolves #731
Resolves #747

### CODE REVIEW

**Review by haraldh — APPROVED** [review]

**Comment by npmccallum** [comment]
@haraldh The `EMSGSIZE` test appears to be failing on your kernel...

**Comment by haraldh** [comment]
> @haraldh The `EMSGSIZE` test appears to be failing on your kernel...

@npmccallum no `sev` passes... the github native container does not have any, so it is using the binary directly, which does not emit `-EMSGSIZE` but happily writes the stuff.

File: enarx-keep/tests/bin/write_emsgsize.c [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#include <sys/syscall.h>
+#include <unistd.h>
+#include <errno.h>
+
+void _start(void) {
+    char msg[128 * 1024];
```
Comment (lkatalin) [inline enarx-keep/tests/bin/write_emsgsize.c:6]: How is this size chosen for "too large"?

Comment (npmccallum) [reply enarx-keep/tests/bin/write_emsgsize.c:6]: The largest single write that I would consider "reasonable" is the ~64k maximum size of a UDP packet.

**Comment by lkatalin** [comment]
Also looks like it resolves #747 , unless there was more we wanted to do with that @npmccallum ?

**Comment by npmccallum** [comment]
This PR requires https://github.com/enarx/enarx/pull/907

**Pull request merged**